kernel/src/features/constraints.rs

```diff
@@ -0,0 +1,18 @@
+/// A constraint in a check constraint
+#[derive(Eq, PartialEq, Debug, Default, Clone)]
+pub struct Constraint {
+    /// The full path to the field.
+    pub name: String,
+    /// The SQL string that must always evaluate to true.
+    pub expr: String,
```

Comment:
hm I need to think about how we will integrate with this (perhaps parsing to `Expression` here but that sounds terrible or passing off opaque to engine?)